tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1437 @@
+    Try {
+      val activationID =
+        if ((result.statusCode == OK) || (result.statusCode == Accepted)) result.getField("activationId") else ""
+      activationID
+    } toOption
+  }
+}
+
+class WskRestAction
+    extends RunWskRestCmd
+    with ListOrGetFromCollectionRest
+    with DeleteFromCollectionRest
+    with HasActivationRest
+    with BaseAction {
+
+  override protected val noun = "actions"
+  override implicit val config = PatienceConfig(100 seconds, 15 milliseconds)
+
+  /**
+   * Creates action. Parameters mirror those available in the REST.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def create(
+    name: String,
+    artifact: Option[String],
+    kind: Option[String] = None, // one of docker, copy, sequence or none for autoselect else an explicit type
+    main: Option[String] = None,
+    docker: Option[String] = None,
+    parameters: Map[String, JsValue] = Map(),
+    annotations: Map[String, JsValue] = Map(),
+    parameterFile: Option[String] = None,
+    annotationFile: Option[String] = None,
+    timeout: Option[Duration] = None,
+    memory: Option[ByteSize] = None,
+    logsize: Option[ByteSize] = None,
+    shared: Option[Boolean] = None,
+    update: Boolean = false,
+    web: Option[String] = None,
+    expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+
+    val (namespace, actName) = getNamespaceActionName(name)
+    var exec = Map[String, JsValue]()
+    var code = ""
+    var kindType = ""
+
+    val (ext, artifactFile) = artifact map { a =>
+      (getExt(a), a)
+    } getOrElse (null, "")
+
+    ext match {
+      case ".jar" => {
+        kindType = "java:default"
+        val jar = FileUtils.readFileToByteArray(new File(artifactFile))
+        code = Base64.getEncoder.encodeToString(jar)
+      }
+      case ".zip" => {
+        val zip = FileUtils.readFileToByteArray(new File(artifactFile))
+        code = Base64.getEncoder.encodeToString(zip)
+      }
+      case ".js" => {
+        kindType = "nodejs:default"
+        code = FileUtils.readFileToString(new File(artifactFile))
+      }
+      case ".py" => {
+        kindType = "python:default"
+        code = FileUtils.readFileToString(new File(artifactFile))
+      }
+      case ".swift" => {
+        kindType = "swift:default"
+        code = FileUtils.readFileToString(new File(artifactFile))
+      }
+      case ".php" => {
+        kindType = "php:default"
+        code = FileUtils.readFileToString(new File(artifactFile))
+      }
+      case _ =>
+    }
+
+    kindType = docker map { d =>
+      "blackbox"
+    } getOrElse kindType
+
+    var (params, annos) = getParamsAnnos(parameters, annotations, parameterFile, annotationFile, web = web)
+
+    val inputKindType = kind map { k =>
```

Comment:
Shouldn't need this as kind is defined as a `Some` or `None`.

Reply:
Done